contracts/Evolution3Serum/Evolution3Serum.sol

```diff
@@ -0,0 +1,267 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+/// @title Pixelmon Evolution 3 Serum Contract
+/// @author LiquidX
+/// @notice This smart contract is for Evolution 3 Serum collection
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/common/ERC2981.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "operator-filter-registry/src/DefaultOperatorFilterer.sol";
+
+/// @notice Thrown when invalid destination address specified address(0)
+error InvalidAddress();
+
+/// @notice Thrown when given input does not meet with requirements of the expected one
+error InvalidInput();
+
+/// @notice Thrown when address is not included in Minter list
+error InvalidMinter();
+
+/// @notice Thrown when address is not included in Burner list
+error InvalidBurner();
+
+/// @notice Thrown when token ID is not match with the existing one
+error InvalidTokenId();
+
+/// @notice Thrown when input is equal to zero
+error ValueCanNotBeZero();
+
+/// @notice Thrown when token supply already reaches its maximum limit
+error MintAmountForTokenIDExceeded();
+
+/// @notice Strings library will be used to convert uint256 to string
+using Strings for uint256;
+
+/// @notice Pixelmon Evolution 3 Serum Smart Contract
+/// @dev This is an ERC1155 token token standard smart contract
+/// @dev Using Ownable modifiers for configuration methods
+contract Evolution3Serum is ERC1155, Ownable, ERC2981, DefaultOperatorFilterer {
```

Comment:
We already discussed this with Lucas.  It will not support those market places that can bypass the contract owner royalty fee. We can not check this on the unit test. we can test this royalty enforcement in the testnet open-sea after deploying it on the goerli network.